tests/http.go

```diff
@@ -13,9 +13,18 @@ import (
 	"testing"
 )
 
-// gatewayUrl safely constructs the API url based on the `gateway_url`
+// gatewayURL safely constructs the API url based on the `gateway_url`
 // in the ENV.
-func gatewayUrl(t *testing.T, reqPath, query string) string {
+func gatewayURL(t *testing.T) string {
+	t.Helper()
+	uri, err := url.Parse(os.Getenv("gateway_url"))
+	if err != nil {
+		t.Fatalf("invalid gateway url %s", err)
+	}
+	return uri.String()
+}
+
+func resourceURL(t *testing.T, reqPath, query string) string {
```

Comment:
can we add a doc string here so that future contributors will know when to use which method?

Also, i think the doc string for `gatewayURL` is incorrect now, it just returns the base URL for the gateway, it doesn't construct an api url anymore